packages/hawtio/src/plugins/springboot/TraceView.tsx

```diff
@@ -0,0 +1,328 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import {
+  Bullseye,
+  Button,
+  CodeBlock,
+  CodeBlockCode,
+  Dropdown,
+  DropdownItem,
+  DropdownToggle,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateIcon,
+  Flex,
+  FormGroup,
+  Label,
+  Modal,
+  PageSection,
+  Pagination,
+  SearchInput,
+  Toolbar,
+  ToolbarContent,
+  ToolbarFilter,
+  ToolbarGroup,
+  ToolbarItem,
+} from '@patternfly/react-core'
+import { TableComposable, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table'
+import { CheckCircleIcon, ExclamationCircleIcon, SearchIcon } from '@patternfly/react-icons'
+import { Trace } from './types'
+import { springbootService } from './springboot-service'
+
+const HttpStatusIcon: React.FunctionComponent<{ code: number }> = ({ code }) => {
+  if (code < 400) return <CheckCircleIcon color={'#3E8635'} />
+  else return <ExclamationCircleIcon color={'#C9190B'} />
+}
+
+const HttpMethodLabel: React.FunctionComponent<{ method: string }> = ({ method }) => {
+  switch (method) {
+    case 'GET':
+    case 'HEAD':
+      return <Label color='blue'>{method}</Label>
+    case 'POST':
+      return <Label color='orange'>{method}</Label>
+    case 'DELETE':
+      return <Label color='red'>{method}</Label>
+    case 'PUT':
+    case 'PATCH':
+      return <Label color='green'>{method}</Label>
+    default:
+      return <Label color='grey'>{method}</Label>
+  }
+}
+
+const TraceDetails: React.FunctionComponent<{
+  isOpen: boolean
+  setIsOpen: (opened: boolean) => void
+  traceInfo: string
+}> = ({ isOpen, setIsOpen, traceInfo }) => {
+  return (
+    <Modal
+      bodyAriaLabel='Trace detail'
+      tabIndex={0}
+      isOpen={isOpen}
+      variant='large'
+      title='Trace'
+      onClose={() => setIsOpen(false)}
+    >
+      <CodeBlock>
+        <CodeBlockCode>{traceInfo}</CodeBlockCode>
+      </CodeBlock>
+    </Modal>
+  )
+}
+export const TraceView: React.FunctionComponent = () => {
+  const [traces, setTraces] = useState<Trace[]>([])
+  const [filteredTraces, setFilteredTraces] = useState<Trace[]>([])
+
+  const [httpMethodFilter, setMethodFilter] = useState<string>('ALL')
+  const [page, setPage] = useState(1)
+  const [perPage, setPerPage] = useState(20)
+  const [searchTerm, setSearchTerm] = useState('')
+  const [filters, setFilters] = useState<string[]>([])
+  const [currentTraceFilter, setCurrentTraceFilter] = useState('Timestamp')
+  const [isFilterDropdownOpen, setIsFilterDropdownOpen] = useState(false)
+  const [isHttpMethodFilterDropdownOpen, setIsHttpMethodFilterDropdownOpen] = useState(false)
+  const [isTraceDetailsOpen, setIsTraceDetailsOpen] = useState(false)
+  const [traceDetails, setTraceDetails] = useState<string>('')
+
+  const handleSearch = useCallback(
```

Comment:
why do we need `useCallback`?

Reply:
So when I used handleSearch in the useEffect it required to be as a dependency. But when it's as dependency handleSearch refference is constantly changing and it ends up in the render roop. By using useCallback handleSearch is recreated only when dependencies changes. But I'm thinking about just removing it and moving the logic directly to the useEffect because I'm not using it anywhere else.